scvi/inference/inference.py

```diff
@@ -42,15 +46,26 @@ class UnsupervisedTrainer(Trainer):
     def __init__(
         self,
         model,
-        gene_dataset,
-        train_size=0.8,
-        test_size=None,
-        n_epochs_kl_warmup=400,
-        normalize_loss=None,
+        gene_dataset: GeneExpressionDataset,
+        train_size: Union[int, float] = 0.8,
+        test_size: Union[int, float] = None,
+        n_iter_kl_warmup: int = None,
+        n_epochs_kl_warmup: int = 400,
+        normalize_loss: bool = None,
         **kwargs
     ):
         super().__init__(model, gene_dataset, **kwargs)
+
+        # Set up number of warmup iterations
+        n_reference_cells = min(10000, gene_dataset.nb_cells)
+        if n_iter_kl_warmup is not None:
```

Comment:
I'm not sure I understand this default. if you set a not None value that should be what is used. I was only suggesting 400 * 10000/batch_size as the default in the function signature.

Reply:
The motivation for `n_reference_cells = min(10000, gene_dataset.nb_cells)` was to take into account the dataset size to set up the threshold iteration number.

Reply:
It's still a bit weird to me that you put a number for this and then it gets changed. So if I set this to be 100 iters, and my dataset is very large, then the trainer changes it to (10000/128) * 100?

I'm suggesting that if you set it to 100 iters that's all it should do. The default I'm suggesting is so that the performance of scVI doesn't change very much if doing iter warmup vs epoch warmup.

Reply:
Ok I think I get it!